packages/extension-api/src/extension-api.d.ts

```diff
@@ -2424,4 +2424,47 @@ declare module '@podman-desktop/api' {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     export function setValue(key: string, value: any, scope?: 'onboarding'): void;
   }
+
+  export interface CliToolOptions {
+    id: string;
+    name: string;
+    displayName: string;
+    markdownDescription: string;
+    images: ProviderImages;
+    location?: string;
+    version?: string;
+    detection?: {
+      versionOptions: string[]; // run command by name using these options
+      versionParser: (string) => string;
+    };
+  }
+
+  export type CliToolState =
+    | 'setup-needed'
+    | 'detection-pending'
+    | 'detecting'
+    | 'installation-pending'
+    | 'installing'
+    | 'update-pending'
+    | 'updating'
+    | 'installed'
+    | 'installed-unknown'
+    | 'uninstall-pending'
+    | 'uninstalling'
+    | 'unknown';
+
+  export interface CliTool extends CliToolOptions, Disposable {
+    extensionInfo: {
+      id: string;
+      label: string;
+    };
+
+    detect(): Promise<void>;
+    state: CliToolState;
+  }
+
+  export namespace cli {
+    export function createCliTool(options: CliToolOptions): CliTool;
+    export function listCliTools(): CliTool[];
```

Comment:
should it return CliTool objects or some 'info' view ? (like a view, not the real object)

Reply:
Should it return proxy, that would let call lifecycle methods like install (if we decide to put them in CliTool), but block any other modification for non-owner extensions?

Another option is just expose current internal method that returns Infos used in UI. Is that what you getting at?

Reply:
for now it's ok as we have mostly 'properties' but yes I would expect that an extension wouldn't be able to call methods of another CLI registered elsewhere

so don't know if proxy works or just different objects that provides only 'properties' (like is there a lifecycle: yes/no or what are the lifecycle methods but there is no direct call possible on these objects)


if it's easier for now, we can remove the list method from the API

Reply:
Let's remove it.